pkg/sentry/identity/kubernetes/validator.go

```diff
@@ -0,0 +1,60 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dapr/dapr/pkg/sentry/identity"
+	kauthapi "k8s.io/api/authentication/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	kauth "k8s.io/client-go/kubernetes/typed/authentication/v1"
+)
+
+const (
+	errPrefix = "csr validation failed"
+)
+
+func NewValidator(client k8s.Interface) identity.Validator {
+	return &validator{
+		client: client,
+		auth:   client.AuthenticationV1(),
+	}
+}
+
+type validator struct {
+	client k8s.Interface
+	auth   kauth.AuthenticationV1Interface
+}
+
+func (v *validator) Validate(id, token string) error {
```

Comment:
We can validate this method using [authentication client fake](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/client-go/kubernetes/typed/authentication/v1/fake) 

Test cases will be:
1. id or token are not given
2. invalid token
3. Authentcated is nil
4. token is not valid.
5. token id mismatched.

Fake can be used for validate your code as a unit-test. that's why k8s provides fakes as far as I know.